tvchannellist/channel.py

```diff
@@ -30,14 +32,16 @@ def load_engine(self, country, zipcode=None):
                 class_ = getattr(module, cls.__name__)
                 self.engine = class_(zipcode)
 
-    def load_channels(self):
+    async def load_channels(self):
         """Load channels."""
-        self.engine.load_channels()
+        async with ClientSession() as session:
```

Comment:
It's good if an aiohttp client session can be passed from the user of the library. Eg Home Assistant wants all libraries that use aiohttp to use the home assistant aiohttp client session.

Probably store the session on the instance, when instantiating the class.

Reply:
Thanks a lot for your contribution, I really appreciate it.
I'll do like that (adding the session to the `ChannelController`'s `__init__` method) so that it can be passed to engines.
PS: Feel free to add an engine for your home country ;)

PPS: Little bit out of topic... as you know I'm new to python... I've seen that there is the possibility to specify at function definition the type of parameters (and return value). Coming from c# I like this feature (that I've seen is also used in HA core components). What's your opinion? Do you suggest to use this syntax?

Reply:
That's typing:
https://docs.python.org/3/library/typing.html

It's nice as it helps people understand the code. It can help you catch bugs and find inconsistencies in the code.

Reply:
Here you are.
I've added also https://github.com/tulindo/tvchannellist/issues/8

Reply:
Looks good!